src/applications/claims-status/utils/helpers.js

```diff
@@ -1068,6 +1070,52 @@ export const isAutomated5103Notice = itemDisplayName => {
   return itemDisplayName === 'Automated 5103 Notice Response';
 };
 
+// Capitalizes the first letter in a given string
+export const sentenceCase = str => {
+  return typeof str === 'string' && str.length > 0
+    ? str[0].toUpperCase().concat(str.substring(1))
+    : '';
+};
+
+// Returns a title for a claim for the specified placement:
+//   'detail' for the heading on the single page view
+//   'breadcrumb' for the breadcrumbs on the single page view
+//   'document' for the browser tab title on the single page view
+//   the default return is for the list view (card heading)
+export const generateClaimTitle = (claim, placement, tab) => {
+  // This will default to 'disability compensation'
+  const claimType = getClaimType(claim).toLowerCase();
+  const isRequestToAddOrRemoveDependent = addOrRemoveDependentClaimTypeCodes.includes(
+    claim?.attributes?.claimTypeCode,
+  );
+  // Determine which word should follow the tab name.
+  // "Files for...", "Status of...", "Details of...", "Overview of..."
+  const joiner = tab === 'Files' ? 'for' : 'of';
+  // Use the following to (somewhat) cut down on repetition in the switch below.
+  const addOrRemoveDependentClaimTitle = 'request to add or remove a dependent';
+  const baseClaimTitle = isRequestToAddOrRemoveDependent
+    ? addOrRemoveDependentClaimTitle
+    : `${claimType} claim`;
+  // This switch may not scale well; it might be better to create a map of the strings instead.
+  // For examples of output given different parameters, see the unit tests.
+  switch (placement) {
+    case 'detail':
+      return `Your ${baseClaimTitle}`;
+    case 'breadcrumb':
+      return `${tab} ${joiner} your ${baseClaimTitle}`;
+    case 'document':
+      if (claimAvailable(claim)) {
+        const formattedDate = buildDateFormatter()(claim.attributes.claimDate);
+        return titleCase(`${tab} ${joiner} ${formattedDate} ${baseClaimTitle}`);
+      }
+      return `${tab} ${joiner} Your Claim`;
+    default:
+      return isRequestToAddOrRemoveDependent
+        ? sentenceCase(addOrRemoveDependentClaimTitle)
+        : `Claim for ${claimType}`;
+  }
```

Comment:
Let me know how y'all feel about the readability here; I could see an argument for it being a bit over-engineered (to reduce repetition).  The alternative here could be something more readable, maybe like this:

```js  // Determine which word should follow the tab name.
// "Files for...", "Status of...", "Details of...", "Overview of..."
const joiner = tab === 'Files' ? 'for' : 'of';
switch (placement) {
  case 'detail':
    return isRequestToAddOrRemoveDependent
      ? 'Your request to add or remove a dependent'
      : `Your ${claimType} claim`;
  case 'breadcrumb':
    return isRequestToAddOrRemoveDependent
      ? `${tab} ${joiner} your request to add or remove a dependent`
      : `${tab} ${joiner} your ${claimType} claim`;
  case 'document':
    if (claimAvailable(claim)) {
      const formattedDate = buildDateFormatter()(claim.attributes.claimDate);
      return isRequestToAddOrRemoveDependent
        ? `${tab} ${joiner} ${formattedDate} Request to Add or Remove a Dependent`
        : titleCase(`${tab} ${joiner} ${formattedDate} ${claimType} Claim`);
    }
    return `${tab} ${joiner} Your Claim`;
  default:
    return isRequestToAddOrRemoveDependent
      ? 'Request to add or remove a dependent'
      : `Claim for ${claimType}`;
}
```